File: src/etch/examples/simpleecho_cs/server/ServerMain.cs
using System;
using System.Collections.Generic;
using System.Text;

using Etch.Transport;
using Etch.Msg;

namespace etch.examples.simpleecho
{
    class ServerMain
    {
        public ListenerHandler lh;

        public ServerMain()
        {
            lh = new MyDefaultListenerHandler(this);
        }

        public class MyDefaultListenerHandler : DefaultListenerHandler
        {
            ServerMain sm;
            MyDeliveryService sds;

            public MyDefaultListenerHandler(ServerMain sm)
            {
                this.sm = sm;
                this.sds = new MyDeliveryService(this);
            }

            protected override Etch.Transport.MessageHandler<MailboxMessageSource> NewMessageHandler()
            {
                return new MyDeliveryService(this);
            }

            protected override ValueFactory NewValueFactory()
            {
                return new ValueFactorySimpleEcho();
            }

            public override string ToString()
            {
                return "MyDefaultListenerHandler";
            }
            
            public class MyDeliveryService : DefaultDeliveryService<StubSimpleEchoServer>
            {
                MyDefaultListenerHandler sdlh;

                public MyDeliveryService(MyDefaultListenerHandler sdlh)
                {
                    this.sdlh = sdlh;
                }

                protected override StubSimpleEchoServer NewStub(MailboxMessageSource src)
                {
                    this.SetMailboxMessageSource(src);
                    RemoteSimpleEchoClient rsc = new RemoteSimpleEchoClient(this);
                    return new StubSimpleEchoServer(new SimpleEchoServerImpl(rsc), null, null);
                }

                public override string ToString()
                {
                    return "MyDeliveryService";
                }
            }

        }

        public static void Main(String[] args)
        {
            ServerMain sm = new ServerMain();
            Listener l = new Listener(sm.lh, 5, "127.0.0.1", 4001, 0);
            l.Start();
            l.WaitUp(4000);
        }
    }
}

File: src/etch/examples/simpleecho/src/etch/examples/simpleecho/ServerMain.java
package etch.examples.simpleecho;

import metreos.core.io.Listener;
import metreos.core.io.ListenerHandler;
import metreos.util.Log;
import etch.bindings.java.msg.ValueFactory;
import etch.bindings.java.transport.DefaultDeliveryService;
import etch.bindings.java.transport.DefaultListenerHandler;
import etch.bindings.java.transport.DefaultMessageHandler;
import etch.bindings.java.transport.MailboxMessageSource;
import etch.bindings.java.transport.MessageHandler;

public class ServerMain 
{
    /**
     * @param args
     * @throws Exception
     */
    public static void main( String[] args ) throws Exception
    {
        Log.addSink( null );
        
        ListenerHandler lh = new DefaultListenerHandler()
        {
            @Override
            protected MessageHandler<MailboxMessageSource> newMessageHandler()
            {
                return new DefaultMessageHandler<StubSimpleEchoServer>()
                {
                    @Override
                    protected StubSimpleEchoServer newStub( MailboxMessageSource src )
                    {
                        DefaultDeliveryService svc = new DefaultDeliveryService( src, null );
                        RemoteSimpleEchoClient rfc = new RemoteSimpleEchoClient( svc );
                        return new StubSimpleEchoServer( new SimpleEchoServerImpl( rfc ), null, null );
                    }

                    @Override
                    public String toString()
                    {
                        return "SimpleEchoServerMessageHandler";
                    }
                };
            }

            @Override
            protected ValueFactory newValueFactory()
            {
                return new ValueFactorySimpleEcho();
            }

            @Override
            public String toString()
            {
                return "SimpleEchoServerListener";
            }
        };
        
        Listener l = new Listener( lh, 5, null, 4001, 0 );
        l.start();
    }
}

